index.ts

```diff
@@ -18,6 +18,15 @@ import { validateNpmName } from "./helpers/validate-pkg";
 import packageJson from "./package.json";
 import { QuestionArgs, askQuestions, onPromptState } from "./questions";
 
+/* Check if the proxy environment variable is set */
```

Comment:
i wouldn't mix `require` and `import` - how about just adding

`import 'global-agent/bootstrap';` add the beginning?

Reply:
import requires type declaration for gloabl-agent. I have added proxy.ts in helper for a cleaner index.ts

Reply:
did you try https://www.npmjs.com/package/@types/global-agent ?